### PR TITLE
return an object after generating schema

### PR DESCRIPTION
**Pull request by asbalderson**
build_from_openapi_schema returns None, so the method also returns None

create the object, populate it, then return the object

### CODE REVIEW

File: django_rest_generator/client.py [hunk 1]
```diff
@@ -270,7 +270,9 @@ def _build_from_openapi_schema(self, schema_file: str = None):
 
     @classmethod
     def build_from_openapi_schema(cls, schema_file, token=None):
-        return cls(token=token)._build_from_openapi_schema(schema_file)
```
Comment (Basdbruijne) [inline django_rest_generator/client.py:273]: This syntax implies that `build_from_openapi_schema` returns a class instance, but `_build_from_openapi_schema` doesn't return anything.

I would do either
```python3
    @classmethod
    def build_from_openapi_schema(cls, schema_file, token=None):
        cls(token=token)._build_from_openapi_schema(schema_file)
```
if you want nothing returned, or end `_build_from_openapi_schema` with

```python3
return self
```

Comment (Basdbruijne) [reply django_rest_generator/client.py:273]: Also, can you add docstrings?

**Review by Basdbruijne — CHANGES_REQUESTED** [review]

File: django_rest_generator/client.py [hunk 2]
```diff
@@ -270,7 +270,9 @@ def _build_from_openapi_schema(self, schema_file: str = None):
...
+        obj = cls(token=token)
+        obj._build_from_openapi_schema(schema_file)
+        return obj
```
Comment (Basdbruijne) [inline django_rest_generator/client.py:275]: Another way to do this is to end `_build_from_openapi_schema` with `return self`. Mainly though, can you add docstrings?

**Review by Basdbruijne — CHANGES_REQUESTED** [review]

File: django_rest_generator/client.py (hunk 2 above)
Comment (asbalderson) [reply django_rest_generator/client.py:275]: I think `return self` doesnt make sense from the private method
you create the object, build the schema, return the object
building the schema should return None, as its just a builder method.


i can see about adding doc strings, its weird to do for a method i didnt write

Comment (Basdbruijne) [reply django_rest_generator/client.py:275]: Yeah, I get that. Maybe @moisesbenzan can help there?

It is a public method that needs to be overridden when using the client (looking at how we use it in weeblv2), so its important that its documented how it should be used.

Comment (asbalderson) [reply django_rest_generator/client.py:275]: It doesn't need to be over written. its just being over written in our use case. from a general use case you should be OK to just use the the generic api client

See: https://github.com/canonical/django-rest-generator/pull/4/files

**Review by Basdbruijne — APPROVED** [review]

**Pull request merged**